lib/index.js

```diff
@@ -37,17 +37,48 @@ class EvervaultClient {
    * @param {String} apiKey 
    * @returns {void}
    */
-  async _overloadHttpsModule(apiKey, tunnelHostname) {
+  async _overloadHttpsModule(apiKey, tunnelHostname, ignoreDomains) {
     let pem = await this.http.getCert();
 
+    if (!ignoreDomains) ignoreDomains = []
+    const cagesHost = new URL(this.config.http.cageRunUrl).host;
+    ignoreDomains.push(cagesHost);
+    let ignoreExact = [];
+    let ignoreEndsWith = [];
+    ignoreDomains.forEach((domain) => {
+        let exact = domain.startsWith('www.') ? domain.slice(4) : domain;
+        ignoreExact.push(exact);
+        ignoreEndsWith.push('.' + exact);
+        ignoreEndsWith.push('@' + exact);
+    });
+
+    const _exactOrEndsWith = (domain, exactDomains, endsWithDomains) => {
+      if (exactDomains.includes(domain)) return true;
+      for (let end of endsWithDomains) {
+        if (domain?.endsWith(end)) return true;
+      }
+      return false;
+    };
```

Comment:
```suggestion
  async _overloadHttpsModule(apiKey, tunnelHostname, ignoreDomains = []) {
    let pem = await this.http.getCert();

    const cagesHost = new URL(this.config.http.cageRunUrl).host;
    ignoreDomains.push(cagesHost);
    let ignoreExact = [];
    let ignoreEndsWith = [];
    ignoreDomains.forEach((domain) => {
        let exact = domain.startsWith('www.') ? domain.slice(4) : domain;
        ignoreExact.push(exact);
        ignoreEndsWith.push('.' + exact);
        ignoreEndsWith.push('@' + exact);
    });

    const _exactOrEndsWith = (domain, exactDomains, endsWithDomains) => {
      if (exactDomains.includes(domain)) return true;
      for (let end of endsWithDomains) {
        if (domain?.endsWith(end)) return true;
      }
      return false;
    };
```

Reply:
Was it just the `if(!ignoreDomains...` line that was changed?